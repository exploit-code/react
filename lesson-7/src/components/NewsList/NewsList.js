import { connect } from 'react-redux';


// Material UI
import { makeStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardContent from '@material-ui/core/CardContent';


const useStyles = makeStyles({
    card: {
        margin: '0 0 20px 0'
    }
  });

const NewsList = ({ news }) => {

    const classes = useStyles();

    return (
        <>
            { news.map(({sectionName, webPublicationDate, webTitle }) => <Card className={classes.card} key={webTitle}>
                <CardActionArea>
                    <CardContent>

                        <Typography gutterBottom variant="h5" component="h2">
                            { webTitle }
                        </Typography>

                        <Typography gutterBottom variant="body2" color="textSecondary" component="p">
                            { sectionName }
                        </Typography>

                        <Typography gutterBottom variant="body2" color="textSecondary" component="p">
                            { webPublicationDate }
                        </Typography>

                    </CardContent>
                </CardActionArea>
            </Card>) }
        </>
        
    )
}


export default NewsList