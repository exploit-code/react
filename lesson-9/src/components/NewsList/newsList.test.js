import NewsList from './NewsList'
import {render, screen} from '@testing-library/react'
import '@testing-library/jest-dom'

const news = [
    {
        apiUrl: "https://content.guardianapis.com/business/live/2021/aug/16/british-manufacturer-ultra-electronics-private-equity-meggitt-cobham-ftse-100-business-live",
        id: "business/live/2021/aug/16/british-manufacturer-ultra-electronics-private-equity-meggitt-cobham-ftse-100-business-live",
        isHosted: false,
        pillarId: "pillar/news",
        pillarName: "News",
        sectionId: "business",
        sectionName: "Business",
        type: "liveblog",
        webPublicationDate: "2021-08-16T07:06:10Z",
        webTitle: "Cobham agrees Ultra deal and Meggitt nears takeover in British aerospace sell-off â€“ business live",
        webUrl: "https://www.theguardian.com/business/live/2021/aug/16/,british-manufacturer-ultra-electronics-private-equity-meggitt-cobham-ftse-100-business-live"
    },
    {
        apiUrl: "https://content.guardianapis.com/world/live/2021/aug/16/coronavirus-live-news-japan-set-to-extend-state-of-emergency-as-sydney-records-deadliest-pandemic-day",
        id: "world/live/2021/aug/16/coronavirus-live-news-japan-set-to-extend-state-of-emergency-as-sydney-records-deadliest-pandemic-day",
        isHosted: false,
        pillarId: "pillar/news",
        pillarName: "News",
        sectionId: "world",
        sectionName: "World news",
        type: "liveblog",
        webPublicationDate: "2021-08-16T07:05:31Z",
        webTitle: "Coronavirus live news: Japan set to extend state of emergency as Sydney records deadliest pandemic day ",
        webUrl: "https://www.theguardian.com/world/live/2021/aug/16/coronavirus-live-news-japan-set-to-extend-state-of-emergency-as-sydney-records-deadliest-pandemic-day"
    }
]

describe('News List test', () => {

    it('matches snapshot online', () => {
        const component = render(<NewsList news={news} />);
        expect(component).toMatchSnapshot();
    })

    it("search for text on the page 'Coronavirus live news'", () => {
        render(<NewsList news={news} />)

        const wrapper = screen.getByText(/Coronavirus live news/i)

        expect(wrapper).toBeInTheDocument()
    })

    it("search for text on the page 'Cobham agrees Ultra deal'", () => {
        render(<NewsList news={news} />)

        const wrapper = screen.getByText(/Cobham agrees Ultra deal/i)

        expect(wrapper).toBeInTheDocument()
    })

})