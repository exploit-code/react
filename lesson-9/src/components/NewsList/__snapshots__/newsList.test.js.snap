// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News List test matches snapshot online 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <button
          class="MuiButtonBase-root MuiCardActionArea-root"
          tabindex="0"
          type="button"
        >
          <div
            class="MuiCardContent-root"
          >
            <h2
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
            >
              Cobham agrees Ultra deal and Meggitt nears takeover in British aerospace sell-off – business live
            </h2>
            <p
              class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
            >
              Business
            </p>
            <p
              class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
            >
              2021-08-16T07:06:10Z
            </p>
          </div>
          <span
            class="MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <button
          class="MuiButtonBase-root MuiCardActionArea-root"
          tabindex="0"
          type="button"
        >
          <div
            class="MuiCardContent-root"
          >
            <h2
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
            >
              Coronavirus live news: Japan set to extend state of emergency as Sydney records deadliest pandemic day 
            </h2>
            <p
              class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
            >
              World news
            </p>
            <p
              class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
            >
              2021-08-16T07:05:31Z
            </p>
          </div>
          <span
            class="MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiCardContent-root"
        >
          <h2
            class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
          >
            Cobham agrees Ultra deal and Meggitt nears takeover in British aerospace sell-off – business live
          </h2>
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
          >
            Business
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
          >
            2021-08-16T07:06:10Z
          </p>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiCardContent-root"
        >
          <h2
            class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
          >
            Coronavirus live news: Japan set to extend state of emergency as Sydney records deadliest pandemic day 
          </h2>
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
          >
            World news
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
          >
            2021-08-16T07:05:31Z
          </p>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
